src/views/Transitioner.js

```diff
@@ -293,6 +316,70 @@ function isSceneActive(scene: NavigationScene): boolean {
   return scene.isActive;
 }
 
+function getRegularAnimation(
+  indexHasChanged,
+  positionHasChanged,
+  progress,
+  position,
+  nextProps,
+  transitionSpec,
+  timing
+) {
+  return indexHasChanged && positionHasChanged
+    ? [
+        timing(progress, {
+          ...transitionSpec,
+          toValue: 1,
+        }),
+        timing(position, {
+          ...transitionSpec,
+          toValue: nextProps.navigation.state.index,
+        }),
+      ]
+    : [];
+}
+
+function getFlipAnimations(
+  indexHasChanged,
+  positionHasChanged,
+  progress,
+  position,
+  nextProps,
+  transitionSpec
+) {
+  const flipFromAnimation =
+    indexHasChanged && positionHasChanged
+      ? [
+          Animated.timing(progress, {
+            ...transitionSpec,
+            toValue: 0.5,
+          }),
+          Animated.timing(position, {
+            ...transitionSpec,
+            toValue: nextProps.navigation.state.index - 0.5,
+          }),
+        ]
+      : [];
+  const flipToAnimation =
+    indexHasChanged && positionHasChanged
+      ? [
+          Animated.timing(progress, {
+            ...transitionSpec,
+            toValue: 1,
+          }),
+          Animated.timing(position, {
+            ...transitionSpec,
+            toValue: nextProps.navigation.state.index,
```

Comment:
For all these animations, can we add below code to improve performance.
useNativeDriver: true

Reply:
@ramapriya2005 we cannot use useNativeDriver: true, because the app will freeze for an unknown reason